File: frontend/src/objects/rider/Rider.ts
import type { RiderId, RiderStatus } from '@/objects/shared/ids'

export interface Rider {
  id: RiderId
  name: string
  phone: string
  realtimeLocation: string
  status: RiderStatus
  totalOrders: number
  rating: number
  station: string
  salary: number
}

File: backend/src/rider/objects/Rider.scala
package delivery.rider.objects

import delivery.shared.objects.{RiderId, RiderStatus}

final case class Rider(
    id: RiderId,
    name: String,
    phone: String,
    realtimeLocation: String,
    status: RiderStatus,
    totalOrders: Int,
    rating: Double,
    station: String,
    salary: Double
)
